pkg/devfile/adapters/common/generic_test.go

```diff
@@ -17,210 +17,232 @@ type mockExecClient struct {
+				{
 					Exec: &common.Exec{Id: commands[0]},
 				},
-				commands[1]: {
+				{
 					Exec: &common.Exec{Id: commands[1]},
 				},
-				commands[2]: {
+				{
 					Composite: &common.Composite{Id: commands[2]},
 				},
 			},
-			composite: common.Composite{
+			cmd: createFrom(common.Composite{
 				Id:       commands[2],
 				Commands: []string{commands[0], "fake-command"},
 				Parallel: true,
-			},
+			}),
 			execClient: fakeExecClient,
 			wantErr:    true,
 		},
 		{
 			name: "Case 7: Nested composite commands",
-			commandsMap: map[string]common.DevfileCommand{
-				commands[0]: {
+			commands: []common.DevfileCommand{
+				{
 					Exec: &common.Exec{Id: commands[0]},
 				},
-				commands[1]: {
+				{
 					Exec: &common.Exec{Id: commands[1]},
 				},
-				commands[2]: {
+				{
 					Composite: &common.Composite{Id: commands[2], Commands: []string{commands[0], commands[1]}},
 				},
-				commands[3]: {
+				{
 					Composite: &common.Composite{Id: commands[3]},
 				},
 			},
-			composite: common.Composite{
+			cmd: createFrom(common.Composite{
 				Id:       commands[3],
 				Commands: []string{commands[0], commands[2]},
 				Parallel: false,
-			},
+			}),
 			execClient: fakeExecClient,
 			wantErr:    false,
 		},
 		{
 			name: "Case 8: Nested parallel composite commands",
-			commandsMap: map[string]common.DevfileCommand{
-				commands[0]: {
+			commands: []common.DevfileCommand{
+				{
 					Exec: &common.Exec{Id: commands[0]},
 				},
-				commands[1]: {
+				{
 					Exec: &common.Exec{Id: commands[1]},
 				},
-				commands[2]: {
+				{
 					Composite: &common.Composite{Id: commands[2], Commands: []string{commands[0], commands[1]}},
 				},
-				commands[3]: {
+				{
 					Composite: &common.Composite{Id: commands[3]},
 				},
 			},
-			composite: common.Composite{
+			cmd: createFrom(common.Composite{
 				Id:       commands[3],
 				Commands: []string{commands[0], commands[2]},
 				Parallel: true,
-			},
+			}),
 			execClient: fakeExecClient,
 			wantErr:    false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := ExecuteCompositeDevfileAction(tt.execClient, tt.composite, tt.commandsMap, compInfo, true, machineoutput.NewNoOpMachineEventLoggingClient())
+			err := adapter(tt.execClient, tt.commands, cif).ExecuteDevfileCommand(tt.cmd, false)
 			if !tt.wantErr == (err != nil) {
 				t.Errorf("expected %v, wanted %v", err, tt.wantErr)
 			}
 		})
 	}
 }
+
+func adapter(fakeExecClient ExecClient, commands []common.DevfileCommand, cif func(command common.DevfileCommand) (ComponentInfo, error)) *GenericAdapter {
+	devObj := devfileParser.DevfileObj{
+		Data: testingutil.TestDevfileData{
+			Commands: commands,
+		},
+	}
+	ctx := AdapterContext{
+		Devfile: devObj,
+	}
+	a := NewGenericAdapter(fakeExecClient, ctx)
+	a.supervisordComponentInfo = cif
+	a.componentInfo = cif
+	return a
+}
+
+func createFrom(composite common.Composite) common.DevfileCommand {
```

Comment:
something more descriptive `createCommandFrom`

Reply:
Does it really make a difference?

Reply:
Just a bit easier to understand what the function is creating from the first glance.